### DISCUSSION TITLE
Examples for custom field types

### DISCUSSION THREAD

**Discussion by phaer** [pocketbase/pocketbase#20] [Q&A] [answered]
Is there documentation available on how to add custom field types for records when using pocketbase as a go framework?
 
I found the Svelte components of core field types in https://github.com/pocketbase/pocketbase/tree/master/ui/src/components/records/fields and serialization and validation logic in https://github.com/pocketbase/pocketbase/blob/master/models/schema/schema_field.go.
But I haven't found a quick example or tutorial describing the steps needed. I'd be willing to work on a PR if someone who worked on pocketbase before could provide a quick checklist.

Love this project! I explored about similar ideas before, seeing such a clean & nice implementation is great :)

**Comment by ganigeorgiev** [comment, marked as answer]
Custom field types are currently not supported. It'll require also changing how the fronend is loaded (its currently a prebuilt spa) and in order to allow users to inject/dynamically load their field's options panel and rendering will have to implement a little more advanced loading system.

For now, I'm no planning adding support for that. **Stabilizing the PocketBase API and working on [Presentator v3](https://github.com/presentator/presentator/issues/183) is with higher priority.**

Do you have any specific field types in mind that you want to see in PocketBase? If their usage could be generalized I don't mind adding them to the supported existing field types list.

**Reply by markadrake** [reply]
Is this the current stance in 2024? Not being able to add custom field types is crippling and makes PocketBase a no-go.

**Reply by ganigeorgiev** [reply]
@markadrake After the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) there will be basic support for registering external fields (including providing your own UI) but there are no ETAs.

> ... makes PocketBase a no-go

That's OK, I'm not trying to sell you anything or even convince you to use PocketBase. I'll copy one of my replies from another thread:

> With that said, PocketBase doesn't try to solve every existing use case out there. While it may sound strange coming from the author of the project, it is fine to use something else if it doesn't work for you. It is simply a tool that I've created for https://github.com/presentator/presentator/issues/183 and there will be always some limitations since I don't plan growing it beyond what I can maintain on my own. There are plenty of other alternatives that may fit better your use case like: Supabase, Nhost, Appwrite, SurrealDB, EdgeDB, Payload, Strapi, Directus, etc.

**Reply by markadrake** [reply]
Thanks for the reply, @ganigeorgiev. Are you accepting contributions towards the goals outlined in your refactoring thread? I've just started learning Go, but I am better equipped to support your UI efforts. For instance, the common library of components that can be reused across extensions. If you like, I could setup a bare-bones Storybook project with components based on the existing UI. Just something to help/support us getting to the extensibility sooner. All the best,

**Comment by phaer** [comment]
Good to know, thanks!

The first one for my specific use-case would be a map widget. Ideally with a way to use leaflet + plugins in the admin area.

**Reply by ganigeorgiev** [reply]
OK, I think the use case for a simple leaflet map/address field is fairly common, so I'll consider adding it in #21.
